### PR TITLE
Reworks nanocalcium injectors to be better.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Renames / reworks nanocalcium.

It now heals organ damage and IB, in addition to bones. 60 organ damage will be healed over 30 seconds.

If the user has less than 40 hp during the organ / ib / healing section, they will heal a decient amount of all damage types.
If they are over 40 hp during this time, they will take slight brute and burn damage, as it is experimental.

Will restore up to 150 units of blood, up to 70%. Not quite a healthy number, but enough they won't keel over and die from blood loss.

Stage 2 crit will be healed during this time. Stage 1 will be healed by the healing.

To make it _slightly_ easier on bald operatives, it will purge up to 10 units of drugs during the first 40 seconds. However, if they use adrenals during stage 2 of the nanomachines, they will probably die still.

Path not changed, but can be if wished.

## Why It's Good For The Game

Nanocaclium is... a thing.

Good for healing bones... and that is it.
Got IB? Still fucked, unless the IB is in the same limb as a broken bone. Then it will be healed.
Not to mention it takes 2 minutes to work.

Now, while it still leaves you just as vunerable during the healing, the healing is useful.
It will fix almost anything (not clone damage, infections, terror eggs, or missing limbs), but it will allow basically whoever it is used on to get back in action, so long as they are not found / killed during those 2 critical minutes. 

However, some basic brute / burn healing will be needed after using the nanocalcium, which should not be too hard for someone to fix.

May change price to 5 tc. Want peoples thoughts first.

## Testing

Spawning people, beating the living shit out of them / myself. confirming it works as intended.
IB gets fixed, bones get fixed, organs heal, blood regens, body heals, but damages if healthy.

## Changelog
:cl:
tweak: Reworked and renamed nanocalcium. Now heals bones, IB, organ damage, and heals out of crit, but will harm those out of crit slightly.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 1]
```diff
@@ -1289,11 +1294,28 @@
 				return ..()
 			else
 				for(var/obj/item/organ/external/E in M.bodyparts)
-					if(E.is_broken())
-						if(prob(50)) // Each tick has a 50% chance of repearing a bone.
+					if(prob(25)) // Each tick has a 50% chance of repearing a bone.
+						if(E.status & (ORGAN_INT_BLEEDING | ORGAN_BROKEN | ORGAN_SPLINTED | ORGAN_DEAD)) //I can't just check for !E.status
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/medicine.dm:1298]: I'm not sure `ORGAN_DEAD` should be included here, if it's completely necrosed I'm not sure it should be repairable

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 2]
```diff
@@ -1289,11 +1294,28 @@
...
 							to_chat(M, "<span class='notice'>You feel a burning sensation in your [E.name] as it straightens involuntarily!</span>")
 							E.rejuvenate() //Repair it completely.
-							break
+				if(ishuman(M))
+					var/mob/living/carbon/human/H = M
+					for(var/obj/item/organ/internal/I in M.internal_organs) // 60 healing to all internal organs.
+						I.heal_internal_damage(4)
+					if(H.blood_volume < BLOOD_VOLUME_NORMAL * 0.7)// If below 70% blood, regenerate 150 units total
+						H.blood_volume += 10
+					for(var/datum/disease/critical/heart_failure/HF in H.viruses)
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/medicine.dm:1307]: Maybe treat shock with this as well? Just so they don't degrade into it if they get unlucky.

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 3]
```diff
@@ -1289,11 +1294,28 @@
 				return ..()
 			else
 				for(var/obj/item/organ/external/E in M.bodyparts)
-					if(E.is_broken())
-						if(prob(50)) // Each tick has a 50% chance of repearing a bone.
+					if(prob(25)) // Each tick has a 50% chance of repearing a bone.
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/medicine.dm:1297]: outdated comment

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 4]
```diff
@@ -1272,15 +1272,20 @@
 			M.AdjustJitter(8 SECONDS)
 			if(prob(10))
 				to_chat(M, "<span class='warning'>Your skin feels hot and your veins are on fire!</span>")
+				update_flags |= M.adjustFireLoss(1 * REAGENTS_EFFECT_MULTIPLIER, FALSE)
+			for(var/I in M.reagents.reagent_list)
+				var/datum/reagent/R = I
+				if(stimulant_list.Find(R.id))
+					M.reagents.remove_reagent(R.id, 0.5) //We will be generous (for nukies really) and purge out the chemicals during this phase, so they don't fucking die during the next phase. Of course, if they try to use adrenals in the next phase, well...
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/medicine.dm:1279]: Why not just typecast in the list?

```suggestion
			for(var/datum/reagent/R in M.reagents.reagent_list)
				if(stimulant_list.Find(R.id))
					M.reagents.remove_reagent(R.id, 0.5) //We will be generous (for nukies really) and purge out the chemicals during this phase, so they don't fucking die during the next phase. Of course, if they try to use adrenals in the next phase, well...
```

File: code/modules/reagents/chemistry/reagents/medicine.dm (hunk 3 above)
Comment (Qwertytoforty) [reply code/modules/reagents/chemistry/reagents/medicine.dm:1297]: whoops, thought I fixed that

File: code/modules/reagents/chemistry/reagents/medicine.dm (hunk 2 above)
Comment (Qwertytoforty) [reply code/modules/reagents/chemistry/reagents/medicine.dm:1307]: healing will fix it, as shock cures above 25 hp

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**